test/integration/EnvironmentSelect.test.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import {
+    EnvironmentSelect,
+    EnvironmentSelectProps,
+} from '../../src/components/EnvironmentSelect';
+import { Environment } from '../../src/components/types';
+import { render, fireEvent, within } from '@testing-library/react';
+import { create } from 'react-test-renderer';
+
+describe('EnvironmentSelect', () => {
+    it('should render without error', () => {
+        const mockedProps = generateProps();
+        render(<EnvironmentSelect {...mockedProps} />);
+        const tree = create(<EnvironmentSelect {...mockedProps} />).toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+
+    it('should render', () => {
+        const mockedProps = generateProps();
+        render(<EnvironmentSelect {...mockedProps} />);
+    });
+
+    it('should change environment', () => {
+        const mockedProps = generateProps();
+        const { getByTestId } = render(<EnvironmentSelect {...mockedProps} />);
```

Comment:
if possible, use `getByText` instead

Reply:
It is not possible in this case but I will be aware of rtl selector order convention.